numpy/polynomial/chebyshev.py

```diff
@@ -1411,6 +1412,111 @@ def chebgrid3d(x, y, z, c):
     return c
 
 
+def chebvalnd(c, *xi):
+    """
+    Evaluate a N-D Chebyshev series at points (x1, x2, ..., xn).
+
+    This function returns the values:
+
+    .. math:: p(x1,x2,...,xn) = \\sum_{i,j,...,k} c_{i,j,...,k} * T_i(x1) * T_j(x2)*...* T_k(xn)
+
+    The parameters `x1`, `x2`, ...., `xn` are converted to arrays only if
+    they are tuples or a lists, otherwise they are treated as a scalars and
+    they must have the same shape after conversion. In either case, either
+    `x1`, `x2`, ..., `xn` or their elements must support multiplication and
+    addition both with themselves and with the elements of `c`.
+
+    If `c` has fewer than N dimensions, ones are implicitly appended to its
+    shape to make it N-D. The shape of the result will be c.shape[3:] +
+    x1.shape.
+
+    Parameters
+    ----------
+    c : array_like
+        Array of coefficients ordered so that the coefficient of the term of
+        multi-degree i,j,...,k is contained in ``c[i,j,...,k]``. If `c` has
+        dimension greater than N the remaining indices enumerate multiple sets of
+        coefficients.
+    x1, x2,..., xn : array_like, compatible object
+        The N dimensional series is evaluated at the points
+        `(x1, x2,...,xn)`, where `x1`, `x2`,..., `xn` must have the same shape.
+        If any of `x1`, `x2`, ..., `xn` is a list or tuple, it is first converted
+        to an ndarray, otherwise it is left unchanged and if it isn't an
+        ndarray it is  treated as a scalar.
+
+    Returns
+    -------
+    values : ndarray, compatible object
+        The values of the multidimensional polynomial on points formed with
+        triples of corresponding values from `x`, `y`, and `z`.
+
+    See Also
+    --------
+    chebval, chebgridnd, chebfitnd
+    """
+    try:
+        xi = np.array(xi, copy=0)
+    except:
+        raise ValueError('x, y, z are incompatible')
```

Comment:
The error message should refer to the general `(x1, x2,...,xn)` shape for the point coordinates.